feat: add file navigation, sass compilation and exporting

jboix opened the pull request:
## Description

Resolves #4  by introducing the following features:

1. File navigation for accessing the latest version of Pillarbox files within the editor
2. Real-time compilation from SCSS to CSS upon editing SCSS files, with immediate updates to the rendered view.
3. Exporting the current work as a ZIP.

<img width="719" alt="image" src="https://github.com/SRGSSR/pillarbox-web-theme-editor/assets/11271371/44fa43ac-ac29-4a1a-a326-beca56747be5">

## Changes Made

- Introduced `SassWorkspaceCompiler` for SASS compilation and virtual workspace management.
- Implemented `VirtualSassImporter` to resolve import directives in a virtual file system.
- Enabled inclusion of static resources in SASS compilation.
- Added a prebuild script to generate virtual workspace structure from `@srgssr/pillarbox-web` SCSS files.
- Introduced `TogglePaneButton` component for theme editor file navigation.
- Added functionality to display and edit SCSS files from a virtual file system in the theme editor.
- Introduces a `ZipWorkspace` service to allow the exporting of the current workspace as a zip.

## Checklist

- [x] I have followed the project's style and contribution guidelines.
- [x] I have performed a self-review of my own changes.
- [x] I have made corresponding changes to the documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.

jboix committed:
feat: add sass compilation and virtual file management

Introduces the SassWorkspaceCompiler to allow SASS compilation and workspace management. This
utility facilitates the compilation of SASS files using a virtual file system, it allows the
inclusion of static resources.

The VirtualSassImporter, implementing the Importer interface, efficiently resolves import directives
within this virtual file system.

***Example***

```javascript
// Define your workspace
const workspace = [
  {
    name: 'main.scss',
    content: '@import "variables"; .body { color: $primaryColor; }',
    type: 'scss'
  },
  {
    name: 'variables.scss',
    content: '$primaryColor: blue;',
    type: 'scss'
  }
];

// Instantiate the compiler with the workspace
const compiler = new SassWorkspaceCompiler(testWorkspace, {}, 'main.scss');

// Compile the SCSS
const compiledCss = compiler.compile();
```

Additionally, a prebuild script has been introduced to automate the generation of the virtual
workspace structure directly from the `@srgssr/pillarbox-web` dependency's SCSS files.

jboix committed:
feat: add toggle pane button for theme editor file navigation

Introduces a new `TogglePaneButton` component, aimed at displaying a virtual file system in order to
edit the multiple `scss` files found in the pillarbox default theme. When a file in the tree view is
selected, its content is loaded into the theme editor for modification and previewing.

***Example***
```html
<toggle-pane-button label="Display a message">
  <span>Hello World!</span>
</toggle-pane-button>
```

jboix committed:
feat: add export button

Introduces a new `ZipWorkspace` service to allow the exporting of the current workspace as a zip.

***Example***

```javascript
// Define your workspace
const workspace = [
  {
    name: 'main.scss',
    content: '@import "variables"; .body { color: $primaryColor; }',
    type: 'scss'
  },
  {
    name: 'variables.scss',
    content: '$primaryColor: blue;',
    type: 'scss'
  }
];

// Instantiate the zip workspace and trigger the download
await new ZipWorkspace(testWorkspace).download('testWorkspace.zip');
```

amtins commented on package.json:
```diff
@@ -4,21 +4,24 @@
   "scripts": {
-    "build": "vite build",
+    "prebuild": "node .prebuild.js",
+    "build": "npm run prebuild && vite build",
```
```suggestion
    "build": "vite build",
```
npm offers `pre` and `post` scripts that will run automatically before or after a script.

amtins commented:
There's one use case that would be interesting to handle, it could be part of a different PR, and that's being able to change the media.

Today I can't change the appearance of subtitles or live layout.

jboix committed:
fix: remove prebuild from build step

npm offers `pre` and `post` scripts that will run automatically before or after a script.

Co-authored-by: André M. <34163393+amtins@users.noreply.github.com>

amtins reviewed (approved):
LGTM!
